api/provider/aws/common.go

```diff
@@ -53,6 +54,26 @@ func describeTaskDefinition(ecsapi ecsiface.ECSAPI, taskDefinitionARN string) (*
 	return output.TaskDefinition, nil
 }
 
+func getLaunchTypeFromEnvironmentID(store tag.Store, environmentID string) (string, error) {
+	var launchType string
+	tags, err := store.SelectByTypeAndID("environment", environmentID)
+	if err != nil {
+		return "", err
+	}
+
+	if tag, ok := tags.WithKey("type").First(); ok {
+		if tag.Value == models.EnvironmentTypeDynamic {
```

Comment:
I think we should throw an error if we don't find the environment's `type` tag. 
Maybe a more conservative approach here could be something like:
```go
if tag, ok := tags.WithKey("type").First(); ok {
    switch tag.Value {
        case models.EnvironmentTypeDynamic:
            return ecs.LaunchTypeFargate, nil
        case models.EnvironmentTypeStatic:
            return ecs.LaunchTypeEc2, nil
        default:
            return "", fmt.Errorf("...")
    }
}

return "", fmt.Errorf("...")
```

Reply:
@zpatrick Oh good - I've been thinking about this recently, actually. I was going back and forth between erroring out or having a default. Do we have a way to retroactively generate environment type tags for existing environments when a user updates?

Reply:
we do have a way to do that, don't we? We have an API for tags, so we can create new arbitrary tags as needed.

Reply:
No we don't retroactively add tags; keep in mind this is a minor version change so we don't need to worry about backwards compatibility

Reply:
✅